staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -207,11 +207,15 @@ type LeaderElector struct {
 // stopped holding the leader lease
 func (le *LeaderElector) Run(ctx context.Context) {
 	defer runtime.HandleCrash()
-	defer le.config.Callbacks.OnStoppedLeading()
 
 	if !le.acquire(ctx) {
 		return // ctx signalled done
 	}
+	defer func() {
```

Comment:
embarrassed ! We can actually complete this operation by moving the defer func to the bottom....